mongodbatlas/fw_data_source_mongodbatlas_alert_configuration.go

```diff
@@ -0,0 +1,478 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/zclconf/go-cty/cty"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+var _ datasource.DataSource = &AlertConfigurationDS{}
+var _ datasource.DataSourceWithConfigure = &AlertConfigurationDS{}
+
+type tfAlertConfigurationDSModel struct {
+	ID                    types.String                      `tfsdk:"id"`
+	ProjectID             types.String                      `tfsdk:"project_id"`
+	AlertConfigurationID  types.String                      `tfsdk:"alert_configuration_id"`
+	EventType             types.String                      `tfsdk:"event_type"`
+	Created               types.String                      `tfsdk:"created"`
+	Updated               types.String                      `tfsdk:"updated"`
+	Matcher               []tfMatcherModel                  `tfsdk:"matcher"`
+	MetricThresholdConfig []tfMetricThresholdConfigModel    `tfsdk:"metric_threshold_config"`
+	ThresholdConfig       []tfThresholdConfigModel          `tfsdk:"threshold_config"`
+	Notification          []tfNotificationModel             `tfsdk:"notification"`
+	Output                []tfAlertConfigurationOutputModel `tfsdk:"output"`
+	Enabled               types.Bool                        `tfsdk:"enabled"`
+}
+
+type tfAlertConfigurationOutputModel struct {
+	Type  types.String `tfsdk:"type"`
+	Label types.String `tfsdk:"label"`
+	Value types.String `tfsdk:"value"`
+}
+
+func NewAlertConfigurationDS() datasource.DataSource {
+	return &AlertConfigurationDS{}
+}
+
+type AlertConfigurationDS struct {
+	client *MongoDBClient
+}
+
+func (d *AlertConfigurationDS) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, alertConfigurationResourceName)
+}
+
+func (d *AlertConfigurationDS) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	client, err := ConfigureClientInResource(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+	d.client = client
+}
+
+var alertConfigDSSchemaBlocks = map[string]schema.Block{
```

Comment:
[q] why does `output` need to be a block instead of an attribute?

Reply:
This is for backwards compatibility. The optional `output` list is defined by the user, this would be an example:
```
data "mongodbatlas_alert_configuration" "test" {
	project_id             = "${mongodbatlas_alert_configuration.test.project_id}"
	alert_configuration_id = "${mongodbatlas_alert_configuration.test.id}"

	output {
		type = "resource_import"
		label = %[3]q
	}
	output {
		type = "resource_hcl"
		label = %[3]q
	}
}
```